src/GovUk.Education.ExploreEducationStatistics.Admin/Services/DataBlockService.cs

```diff
@@ -116,7 +127,24 @@ public class DataBlockService : IDataBlockService
                     {
                         var dataBlocks = await _releaseContentBlockRepository.GetAll<DataBlock>(release.Id);
 
-                        return _mapper.Map<List<DataBlockSummaryViewModel>>(dataBlocks)
+                        return (await dataBlocks.SelectAsync(async block =>
+                            {
+                                var inContent = block.ContentSectionId != null
+                                                || await _context.KeyStatisticsDataBlock.AnyAsync(
+                                                    ks => ks.DataBlockId == block.Id);
+                                return new DataBlockSummaryViewModel
```

Comment:
> Do we not have an AutoMapper rule for this?

We did but I removed it. I originally wrote this to use `AutoMapper`, but didn't like it. 

It looked like this 
```
                                var result = _mapper.Map<DataBlockSummaryViewModel>(block);
                                result.InContent = block.ContentSectionId != null
                                                   || await _context.KeyStatisticsDataBlock.AnyAsync(
                                                       ks => ks.DataBlockId == block.Id);
                                return result;
```

The mapping was already working out `ChartsCount`, and it couldn't do `InContent` so that needed to be added after it was mapped. This working out of things in different places seemed to defeat the point of using AutoMapper - and leads to the potential for someone to use the mapping and not realise they're still responsible for setting `InContent`. It also means that when you `Map` called, you don't necessarily know what it's doing - is it setting an order? etc.

This was the instance was what led me to raise moving away from AutoMapper in one of our dev meetings. I discussed this specific instance with someone - I think Ben - and this is what we arrived at.

I'm not sure what I think is best with this. I prefer the code as it is compared to using AutoMapper. My gut wants to create our own mappings with something like `_mappingService.ToDataBlockSummaryViewModel(dataBlock)` - but then how would we ensure this gets used?

Reply:
Fine by me!